make generate/format target does not work on darwin

In https://github.com/coreos/prometheus-operator/pull/3284 I had to run `make generate` to update the Documentation/compatibility.md file to include the new versions of Prometheus. The `make generate` (even with running `make clean`) does not produce anything. In CI there is a failure. When adding the new versions manually to the file, and running `make --always-make generate ` or `make generate` produces no new changes, so it does not remove the added fields. 

`make format` fails for me with the following error, but seems unrelated to the above:
```
GOOS= GOARCH=amd64 go run -v ./scripts/generate-crds.go --controller-gen=/Users/lili/go/src/github.com/coreos/prometheus-operator/tmp/bin/controller-gen --gojsontoyaml=/Users/lili/go/src/github.com/coreos/prometheus-operator/tmp/bin/gojsontoyaml
Installing tools from scripts/tools.go
go install -mod=readonly -modfile=scripts/go.mod github.com/brancz/gojsontoyaml
go install -mod=readonly -modfile=scripts/go.mod github.com/campoy/embedmd
go install -mod=readonly -modfile=scripts/go.mod github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb
go install -mod=readonly -modfile=scripts/go.mod github.com/google/go-jsonnet/cmd/jsonnet
go install -mod=readonly -modfile=scripts/go.mod github.com/google/go-jsonnet/cmd/jsonnetfmt
go install -mod=readonly -modfile=scripts/go.mod k8s.io/code-generator/cmd/client-gen
go install -mod=readonly -modfile=scripts/go.mod k8s.io/code-generator/cmd/informer-gen
go install -mod=readonly -modfile=scripts/go.mod k8s.io/code-generator/cmd/lister-gen
go install -mod=readonly -modfile=scripts/go.mod sigs.k8s.io/controller-tools/cmd/controller-gen
sed -i                                                            \
		"s/prometheusOperator: 'v.*',/prometheusOperator: 'v0.39.0',/" \
		jsonnet/prometheus-operator/prometheus-operator.libsonnet;
sed: 1: "jsonnet/prometheus-oper ...": invalid command code j
make: *** [jsonnet/prometheus-operator/prometheus-operator.libsonnet] Error 1
```


cc @paulfantom

--- a/Makefile
+++ b/Makefile
@@ -158,9 +158,11 @@ $(RBAC_MANIFESTS): scripts/generate/vendor scripts/generate/prometheus-operator-
 	scripts/generate/build-rbac-prometheus-operator.sh
 
 jsonnet/prometheus-operator/prometheus-operator.libsonnet: VERSION
-	sed -i                                                            \
+	# note: use temporary file to preserve compatibility with darwin
+	sed -i.bak \
 		"s/prometheusOperator: 'v.*',/prometheusOperator: 'v$(VERSION)',/" \
 		jsonnet/prometheus-operator/prometheus-operator.libsonnet;
+	rm jsonnet/prometheus-operator/prometheus-operator.libsonnet.bak
 
 FULLY_GENERATED_DOCS = Documentation/api.md Documentation/compatibility.md
 TO_BE_EXTENDED_DOCS = $(filter-out $(FULLY_GENERATED_DOCS), $(shell find Documentation -type f))

